lib/KafkaBacklogMetrics.js

```diff
@@ -69,9 +69,15 @@ class KafkaBacklogMetrics extends EventEmitter {
     }
 
     isReady() {
-        return this._zookeeper &&
+        if (!(this._zookeeper &&
             this._zookeeper.getState().code ===
-            zookeeper.State.SYNC_CONNECTED.code;
+            zookeeper.State.SYNC_CONNECTED.code)) {
+            this._log.error('KafkaBacklogMetrics is not ready, zookeeper not connected',
+                { error: this._zookeeper.getState().code, state: zookeeper.State });
```

Comment:
```suggestion
                { state: this._zookeeper.getState() });
```

No need to log the State variable I guess, it will contain everything, instead, you can log the current full state and remove the `code` as it will be part of the payload.

Reply:
476b6a5c5e0437a6d8354436b3ecc94167282bf4